src/Filters/Converter.php

```diff
@@ -389,7 +389,7 @@ public function createFacetedSearchFiltersFromQuery(ProductSearchQuery $query)
                 case self::TYPE_CATEGORY:
                     if (isset($facetAndFiltersLabels[$filterLabel])) {
                         foreach ($facetAndFiltersLabels[$filterLabel] as $queryFilter) {
-                            $categories = Category::searchByName($idLang, $queryFilter, true);
+                            $categories = Category::searchByNameAndParentCategoryId($idLang, $queryFilter, (int) $query->getIdCategory());
```

Comment:
Is it ok when using the "All products" page?

Reply:
No it does not work on "all products" if the leveldepth for sub categories makes all the duplicate names categories show up in the list.
But to solve that, the "encodedFacets" have to be changed from "type-name" to "type-id". That would proboboly affect SEO?

Reply:
Yes, that's the problem, I was against using type name, but it's how it started :(

Reply:
At least it fixes one use case, we can improve later :P